clients/fides-js/src/lib/preferences.ts

```diff
@@ -253,6 +275,11 @@ export const updateConsent = async (
     validation?: UpdateConsentValidation;
   },
   consentMethod: ConsentMethod = ConsentMethod.SCRIPT,
+  eventExtraDetails: FidesEventExtraDetails = {
+    trigger: {
+      origin: "external",
```

Comment:
Can we use enum?

Reply:
I considered using enums, but it causes problems with the Tying in the Docs (you have to constantly switch between strings and enums to support the Docs...we have this issue in other places). We don't use enums for the Fides Event names, the trigger type, etc. already so I opted to do this for consistency. It's not a strong opinion I have though so open to push back.

Reply:
Decided to go ahead and use enum for _setting_ `target.type` and `target.origin` but not in the Docs. This seems to be working well.